ci-operator/step-registry/dpu-operator/e2e-tests-workflow/tests/dpu-operator-e2e-tests-workflow-tests-commands.sh

```diff
@@ -1,79 +1,96 @@
-            			echo "Exiting- job has not started for longer than 5 days..."
-            			exit 1
-        		fi
-        		sleep_counter=$((sleep_counter + 60))
-        		sleep 60
-    		fi
+		curl -k --resolve "$endpoint_resolve" -X GET $1/api/json > $JSON_FILE
+		blocked=$(cat $JSON_FILE | jq .blocked)
+		if [[ "$blocked" == "true" ]]; then
+			echo "Job is blocked, waiting for job to start"
+		elif [[ "$blocked" == "false" ]]; then
+				cat $JSON_FILE | jq -r .executable.url
+				break
+		else
+			echo "Error: unknown value of blocked variable: $blocked"
+			exit 1
+		fi
+
+		if [ "$sleep_counter" -ge "$max_sleep_duration" ]; then
+			echo "Exiting- job has not started for longer than 5 days..."
+			exit 1
+		fi
+		sleep_counter=$((sleep_counter + 60))
+		sleep 60
 	done
 }
 
 wait_for_job_to_finish_running() {
-	job_url_id="https://${endpoint}/job/$test_name/$ID"
-	max_sleep_duration=21600  # Maximum sleep duration in seconds (6 hours)
-	sleep_counter=0
+	echoerr "Waiting for job to finish..."
+	local job_url=$1/api/json
+	local max_sleep_duration=21600  # Maximum sleep duration in seconds (6 hours)
+	local sleep_counter=0
+
+	endpoint_resolve="${endpoint}:443:10.0.180.88"
+
 	while :
-	do	
+	do
 		JSON_FILE=$(mktemp /tmp/output.XXX)
-		curl -k -s --resolve "$endpoint_resolve" "${job_url}/api/json" > $JSON_FILE
+		curl -k -s --resolve "$endpoint_resolve" "$job_url" > "$JSON_FILE"
 
-    		# Extract the result field
-    		result=$(cat $JSON_FILE | jq -r '.result')
+		# Extract the result field
+		result=$(jq -r '.result' < "$JSON_FILE")
 
-    		if [[ "$result" != "null" ]]; then
-        		# Job has completed
-        		echo "Job Result: $result"
-	
-			curl_info=$(curl -k -s --resolve "$endpoint_resolve" "${job_url_id}/consoleText")
+		if [[ "$result" != "null" ]]; then
+			# Job has completed
+			echo "Job Result: $result"
+			curl_info=$(curl -k -s --resolve "$endpoint_resolve" "${job_url}/consoleText")
 			echo "$curl_info"
-        
 			if [ "$result" == "SUCCESS" ]; then
-            			exit 0
-        		else
-            			exit 1
-        		fi
-    		else
-        		if [ "$sleep_counter" -ge "$max_sleep_duration" ]; then
-            			echo "Exiting due to long sleep duration..."
-            			exit 1
-        		fi
-        		sleep_counter=$((sleep_counter + 60))
-        		sleep 60
-    		fi
+				exit 0
+			else
+				exit 1
+			fi
+		else
+			if [ "$sleep_counter" -ge "$max_sleep_duration" ]; then
+				echo "Exiting due to long sleep duration..."
+				exit 1
+			fi
+			sleep_counter=$((sleep_counter + 60))
+			sleep 60
+		fi
 	done
 }
 
-trigger_build() {
-	curl -k --resolve "${endpoint_resolve}" "https://${endpoint}/view/dpu-test/job/${test_name}/buildWithParameters?token=$token_dpu_operator_key&pullnumber=$PULL_NUMBER"
-}
-
-token_dpu_operator_key=$(cat "/var/run/token/dpu-token/dpu-key")
 endpoint=$(cat "/var/run/token/dpu-token/url")
+dpu_token=$(cat "/var/run/token/dpu-token/dpu-key")
 test_name="99_Lab217_E2E_IPU_Deploy"
```

Comment:
This is a static string. We may not want to pass it into the function `trigger_build()`